endgame_postprocessing/model_wrappers/sch/testRun.py

```diff
@@ -0,0 +1,23 @@
+import numpy as np
+import pandas as pd
+
+
+def first_row(df):
+    return df.iloc[0]
+
+
+def recombine_ages(raw_data: pd.DataFrame):
+    each_intensity_measure = [
+        df for _, df in raw_data.groupby(["Time", "intensity", "measure"])
+    ]
+    aggregate_functions = {c: first_row for c in raw_data.columns}
+    aggregate_functions["draw_1"] = np.sum
+    aggregate_functions["age_start"] = np.min
+    aggregate_functions["age_end"] = np.max
+
+    aggregated_by_group = [
+        measure_values.aggregate(aggregate_functions)
+        for measure_values in each_intensity_measure
+    ]
+
+    return pd.concat(aggregated_by_group, axis=1).T
```

Comment:
TLDR; just change this to `return pd.DataFrame(aggregated_by_group)`

I believe the issue you are having with the dtypes is because you concat and transpose. Concatinating the list together first will cause a long dataframe first (i.e 2 columns, 5 rows) and pandas pandas to override all the dtypes to `object` if at least one of the values is an object (which there is for intensity/measure).

The output of line 18-21 actually is a list of dictionaries, so you could just do 

`return pd.DataFrame(aggregated_by_group)` and you will be good!